ui/imports/shared/popups/CommunityIntroDialog.qml

```diff
@@ -18,64 +18,147 @@ import SortFilterProxyModel 0.2
 StatusStackModal {
     id: root
 
-    property string name
+    property bool isEditMode: false
+
+    required property string communityName
+    required property string communityIcon
+    required property bool requirementsCheckPending
+
     property string introMessage
     property int accessType
-    property url imageSrc
+
     property bool isInvitationPending: false
-    property int loginType: Constants.LoginType.Password
 
     required property var walletAccountsModel // name, address, emoji, colorId
-    property var walletAssetsModel
+    required property var walletAssetsModel
     required property var permissionsModel  // id, key, permissionType, holdingsListModel, channelsListModel, isPrivate, tokenCriteriaMet
     required property var assetsModel
     required property var collectiblesModel
 
-    required property bool requirementsCheckPending
-
     property var keypairSigningModel
 
+    property var oldSharedAddresses: []
+    onOldSharedAddressesChanged: d.reEvaluateModels()
+    property string oldAirdropAddress: ""
+    onOldAirdropAddressChanged: d.reEvaluateModels()
+
     property var getCurrencyAmount: function (balance, symbol){}
 
+    property var canProfileProveOwnershipOfProvidedAddressesFn: function(addresses) { return false }
+
+    readonly property bool profileProvesOwnershipOfSelectedAddresses: {
+        d.selectedSharedAddressesMap // needed for binding
+        let obj = d.getSelectedAddresses()
+        return root.canProfileProveOwnershipOfProvidedAddressesFn(obj.addresses)
+    }
+
+    readonly property bool allAddressesToRevealBelongToSingleKeypairAndProfileKeypairCanNotProveOwnershipForThem: {
```

Comment:
Yeah, I agree! For me, this naming, although it's self explanatory, it's too much ;)

Reply:
I agree, but I like that the name of the variable describes its purpose, I would really appreciate it if you could suggest a shorter name for this one, but to remain clear about what it does, cause it seems I'm bad at naming variables? :)

Reply:
Even if name contains only hint of what the var if for, when additional comment is provided it's (at least for me) easier to read comment as a normal sentence instead of deciphering camelcases ;) Maybe `singleNonProfileKeypairUsed`? Naming... don't expect us to do the hardest part for you ;)

Reply:
Shortened as much as I could. ;)

Reply:
You can go one step further `allAddressesToRevealBelongToSingleNonProfileKeypair` -> `allAddressesBelongToSingleNonProfileKeypair` in context of this component all addresses are intended to be revealed so I think it can be skipped.

E.g. one property above is `profileProvesOwnershipOfSelectedAddresses` with no `ToReveal` addition.

Reply:
Yes, will skip it, cause we're maintaining current addresses as well, and currently selected addresses and newly selected addresses that can be revealed may differ, this name breaks that confusion. ;)